### PR TITLE
Trunk-5739:Expand updateOrderFulfillerStatus to also set accession number

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/TRUNK-5739

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 1]
```diff
@@ -1253,6 +1253,24 @@ public void updateOrderFulfillerStatus_shouldEditFulfillerStatusInOrder() {
...
+	/**
+	 * @see OrderService#updateOrderFulfillerStatus(org.openmrs.Order,
+	 *      Order.FulfillerStatus, String, String)
+	 */
+	@Test
+	public void updateOrderFulfillerStatus_shouldEditFulfillerStatusWithAccessioNumberInOrder() {
+		Order originalOrder = orderService.getOrder(111);
+		String commentText = "We got the new order";
+		String accessionNumber = "12345";
+		assertNotEquals(originalOrder.getAccessionNumber(), 0);
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:1265]: Minor point, but shouldn't we really be testing here that the accession number is not "12345"?

File: api/src/main/java/org/openmrs/api/OrderService.java [hunk 2]
```diff
@@ -292,6 +292,23 @@
 	@Authorized(PrivilegeConstants.EDIT_ORDERS)
 	public Order updateOrderFulfillerStatus(Order order, Order.FulfillerStatus orderFulfillerStatus, String fullFillerComment);
 	
+	/**
+	 * Updates the fulfillerStatus of an order and the related comment and finally persists it
+	 *
+	 * @param order order whose fulfillerStatus should be changed
+	 * @param orderFulfillerStatus describes the new Order.FulfillerStatus the order should be set to
+	 * @param fullFillerComment is a string which describes a comment that is set while changing the FulfillerStatus 
+	 * @param accessionNumber is the accession number of the order
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/api/OrderService.java:301]: should be "is the accession number to set"

File: api/src/main/java/org/openmrs/api/OrderService.java [hunk 3]
```diff
@@ -292,6 +292,23 @@
...
 	public Order updateOrderFulfillerStatus(Order order, Order.FulfillerStatus orderFulfillerStatus, String fullFillerComment);
 	
+	/**
+	 * Updates the fulfillerStatus of an order and the related comment and finally persists it
+	 *
+	 * @param order order whose fulfillerStatus should be changed
+	 * @param orderFulfillerStatus describes the new Order.FulfillerStatus the order should be set to
+	 * @param fullFillerComment is a string which describes a comment that is set while changing the FulfillerStatus 
+	 * @param accessionNumber is the accession number of the order             
+	 * @return the Order that is updated with an according fulfillerStataus and fulFillerComment
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/api/OrderService.java:302]: ... and accession number"

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java (hunk 1 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/OrderServiceTest.java]: @mogoodrich sure..i have fixed it

File: api/src/main/java/org/openmrs/api/OrderService.java (hunk 2 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/api/OrderService.java]: it done

File: api/src/main/java/org/openmrs/api/OrderService.java (hunk 3 above)
Comment (gitcliff) [reply api/src/main/java/org/openmrs/api/OrderService.java]: its done to here

**Pull request merged**